### PR TITLE
TakeFlashloan action support for DPM Proxy

### PR DESCRIPTION
**Pull request by fenris85**

### CODE REVIEW

File: contracts/dpm/AccountFactory.sol [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.15;
```
Comment (adamskrodzki) [inline contracts/dpm/AccountFactory.sol]: use interfaces and fork in tests, best way to ensure that right version is used

**Review by adamskrodzki — CHANGES_REQUESTED** [review]

File: contracts/libs/DS/ProxyPermission.sol [hunk 2]
```diff
@@ -5,6 +5,7 @@ pragma solidity ^0.8.15;
 import "./DSGuard.sol";
 import "./DSAuth.sol";
 
+import "hardhat/console.sol";
```
Comment (peculiarity) [inline contracts/libs/DS/ProxyPermission.sol:8]: Remove that

File: contracts/actions/common/TakeFlashloan.sol [hunk 3]
```diff
@@ -25,16 +27,21 @@ contract TakeFlashloan is Executable, ProxyPermission {
...
    * @param data Encoded calldata that conforms to the FlashloanData struct
    */
   function execute(bytes calldata data, uint8[] memory) external payable override {
     FlashloanData memory flData = parseInputs(data);
 
     address operationExecutorAddress = registry.getRegisteredService(OPERATION_EXECUTOR);
 
-    if (flData.dsProxyFlashloan) {
-      givePermission(operationExecutorAddress);
+    if (flData.isProxyFlashloan) {
```
Comment (zerotucks) [inline contracts/actions/common/TakeFlashloan.sol:37]: Purely stylistic but I prefer less nesting.

 if (flData.isProxyFlashloan && flData.isDPMProxy) {       IAccountGuard(IAccountImplementation(address(this)).guard()).permit(operationExecutorAddress, address(this), false);
}
if (flData.isProxyFlashloan && !flData.isDPMProxy) {    
 
        removePermission(operationExecutorAddress);
    }

And same below

File: test/aave/Open-dpm.test.ts [hunk 4]
```diff
@@ -0,0 +1,337 @@
+import { JsonRpcProvider } from '@ethersproject/providers'
```
Comment (zerotucks) [inline test/aave/Open-dpm.test.ts]: Can we rename this test to OpenDpm.test.ts to match up with other tests

**Review by zerotucks — APPROVED** [review]

File: contracts/actions/common/TakeFlashloan.sol (hunk 3 above)
Comment (fenris85) [reply contracts/actions/common/TakeFlashloan.sol:37]: resolved by Georgi's change

File: package.json [hunk 5]
```diff
@@ -97,7 +97,8 @@
     "lodash": "^4.17.21",
     "node-cache": "^5.1.2",
     "ramda": "^0.28.0",
-    "typechain": "^8.1.0"
+    "typechain": "^8.1.0",
+    "@openzeppelin/contracts": "4.7.3"
```
Comment (peculiarity) [inline package.json:101]: Are you using these anywhere?

File: test/aave/Open-dpm.test.ts (hunk 4 above)
Comment (peculiarity) [reply test/aave/Open-dpm.test.ts]: @fenris85  Can you do that please?

Comment (fenris85) [reply test/aave/Open-dpm.test.ts]: done

File: package.json (hunk 5 above)
Comment (fenris85) [reply package.json:101]: yes, dpm is using this for cloning

Comment (peculiarity) [reply package.json:101]: Ah, you put the contracts in our repo as well. Alright

**Review by peculiarity — APPROVED** [review]
Looks good to me !

**Pull request merged**